lms/lmsdb/models.py

```diff
@@ -143,6 +153,46 @@ def random_password(cls, stronger: bool = False) -> str:
     def get_notifications(self) -> Iterable['Notification']:
         return Notification.fetch(self)
 
+    def select_notes_of_user(self) -> List[Dict[str, Any]]:
+        staff_roles = (
+            (current_user.role == Role.get_staff_role())
+            | (current_user.role == Role.get_admin_role())
+        )
+        private_notes = (
+            (Note.privacy == NotePublicity.PRIVATE.value)
+            & (Note.creator == current_user.id)
+        )
+        staff_notes = (
+            (Note.privacy == NotePublicity.STAFF.value) & (staff_roles)
+        )
+        user_notes = (
+            (Note.privacy == NotePublicity.USER.value)
+            & ((Note.user == current_user.id) | (staff_roles))
+        )
+        public_notes = Note.privacy == NotePublicity.PUBLIC.value
+
+        return list(Note.select().where(
+            (Note.user == self)
+            & (
+                (private_notes)
+                | (staff_notes)
+                | (user_notes)
+                | (public_notes)
+            ),
+        ).dicts())
+
+    @property
+    def notes(self) -> Iterable[Dict[str, Any]]:
+        notes = self.select_notes_of_user()
+
+        for note in notes:
```

Comment:
It would be better to add it to the query instead of creating 3 more queries

Reply:
I think I can't do it -
In order to get the creator name and not the id.
In order to get the note text and not the id.
In order to get the real date and not the datetime.